rcl/include/rcl/init_options.h

```diff
@@ -34,6 +34,15 @@ typedef struct rcl_init_options_t
 {
   /// Implementation specific pointer.
   struct rcl_init_options_impl_t * impl;
+
+  /// If set, then this value overrides the ROS_DOMAIN_ID environment variable.
+  /**
+   * It defaults to RCL_NODE_OPTIONS_DEFAULT_DOMAIN_ID, which will cause the
+   * context to use the ROS domain ID set in the ROS_DOMAIN_ID environment
+   * variable, or on some systems 0 if the environment variable is not set.
+   */
+  size_t domain_id;
```

Comment:
why do we need this?
`impl->rmw_init_options.domain_id` already have this information.
Consider adding a getter and setter function instead of an extra member

Reply:
okay i will add set/get to rmw_init_options.